tools/harness-thci/ARM.py

```diff
@@ -1580,25 +1579,102 @@ def startNativeCommissioner(self, strPSKc='GRLpassWord'):
+        print '%s call joinCommissioned' % self.port
+        self.__sendCommand('ifconfig up')
+        self.__sendCommand('panid 0x6666')
+        cmd = 'joiner start %s %s' %(strPSKd, self.provisioningUrl)
+        print cmd
+        self.__sendCommand(cmd)
+        time.sleep(90)
+        self.__sendCommand('thread start')
+        time.sleep(30)
+
+#    def getCommissioningLogs(self):
+#       try:
+#         if not self.__logThreadRunning:
+#            self.__logThread = ThreadRunner.run(target = self.__readCommissioningLogs, args = (120,))
+#         rawLogsQ = self.__logThread.get()
+#         ProcessedLogs = []
+#         payload = []
+#         while not rawLogsQ.empty():
+#            EncryptedPacket = PlatformDiagnosticPacket()
+#            rawLogEach = rawLogsQ.get()
+#            print rawLogEach
+#            strParsing1 = rawLogEach.replace('[INFO][THCI]: ','')
+#            logSplitedList = strParsing1.split('|')
+#            for eachPart in logSplitedList:
+#               strParsing2 = eachPart.split('-')
+#               logType = strParsing2[0].strip()
+#               logValue = strParsing2[1].strip()
+#               if 'Direction' in logType:
+#                  EncryptedPacket.Direction = PlatformDiagnosticPacket_Direction.IN if 'recv' in logValue \
+#                     else PlatformDiagnosticPacket_Direction.OUT if 'sent' in logValue \
+#                     else PlatformDiagnosticPacket_Direction.UNKNOWN
+#               elif 'EUI' in logType:
+#                  EncryptedPacket.EUI = self.__MactoHex(logValue)
+#               elif 'IID' in logType:
+#                  EncryptedPacket.IID = self.__MactoHex(logValue)
+#               elif 'Type' in logType:
+#                  EncryptedPacket.Type = PlatformDiagnosticPacket_Type.JOIN_FIN_req if 'JOIN_FIN.req' in logValue \
+#                     else PlatformDiagnosticPacket_Type.JOIN_FIN_rsp if 'JOIN_FIN.resp' in logValue \
+#                     else PlatformDiagnosticPacket_Type.JOIN_ENT_req if 'JOIN_ent.req' in logValue \
+#                     else PlatformDiagnosticPacket_Type.JOIN_ENT_rsp if 'JOIN_ent.resp' in logValue \
+#                     else PlatformDiagnosticPacket_Type.UNKNOWN
+#               elif 'Length' in logType:
+#                  EncryptedPacket.TLVsLength = int(logValue)
+#               elif 'Payload' in logType:
+#                  if logValue != '<null>':
+#                     payloadRaw = logValue.split(':')
+#                     payload = [int(eachOctat,16) for eachOctat in payloadRaw]
+#               else:
+#                  print logType,logValue
+#
+#            EncryptedPacket.TLVs = PlatformPackets.read(EncryptedPacket.Type,payload) if payload != [] else []
+#            ProcessedLogs.append(EncryptedPacket)
+#         return ProcessedLogs
+#      except Exception,e:
+#         ModuleHelper.WriteIntoDebugLogger('ARM.py getCommissioningLogs() '+str(e))
 
     def getCommissioningLogs(self):
-        pass
```

Comment:
With the workaround here, it will help to pass all commissioner tests except 8.1.6 Joiner.